### PR TITLE
[ncp] add Thread Dataset related spinel properties and their get/set handlers in NCP

### PR DESCRIPTION
**Pull request by abtink**
This commit adds new spinel properties related to Thread Operational
Dataset and implements the corresponding get/set handlers for the
new properties in `NcpBase`.

Related PR in wpantund: https://github.com/openthread/wpantund/pull/251

### CODE REVIEW

File: src/ncp/ncp_base_ftd.cpp [hunk 1]
```diff
@@ -592,6 +594,230 @@ void NcpBase::HandleTmfProxyStream(otMessage *aMessage, uint16_t aLocator, uint1
...
+        }
+
+        case SPINEL_PROP_NET_NETWORK_NAME:
+        {
+            const char *name;
+            size_t len;
+
+            SuccessOrExit(error = mDecoder.ReadUtf8(name));
+            len = strlen(name);
+            VerifyOrExit(len <= OT_NETWORK_NAME_MAX_SIZE, error = OT_ERROR_INVALID_ARGS);
```
Comment (jwhui) [inline src/ncp/ncp_base_ftd.cpp]: You seem to use `OT_ERROR_INVALID_ARGS` and `OT_ERROR_PARSE` interchangeably here.  Is that intentional?

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 2]
```diff
@@ -352,3 +352,159 @@ Data per item is:
 *  `C`: Link Quality Out
 *  `C`: Age (seconds since last heard)
 *  `b`: Link established with Router ID or not.
+
+### PROP 5400: SPINEL_PROP_THREAD_ACTIVE_DATASET (#prop-thread-active-dataset)
+
+  * Type: Read-Write
+  * Packing-Encoding: `A(t(iD))`
+
+This property provides access to current Thread Active Operational Dataset.
```
[the remark this answers is not in the document]
Comment (abtink) [reply doc/spinel-protocol-src/spinel-tech-thread.md]: Thanks. Updated the comments.

File: src/ncp/ncp_base_ftd.cpp (hunk 1 above)
Comment (abtink) [reply src/ncp/ncp_base_ftd.cpp]: No. Thanks. Changed this to return `OT_ERROR_INVALID_ARGS` if there is an incorrect value and `OT_ERROR_PARSE` only for parsing errors.

File: src/ncp/ncp_base_ftd.cpp [hunk 3]
```diff
@@ -592,6 +594,230 @@ void NcpBase::HandleTmfProxyStream(otMessage *aMessage, uint16_t aLocator, uint1
...
+    error = otDatasetSendMgmtActiveSet(mInstance, &dataset, extraTlvs, extraTlvsLength);
+
+exit:
+    return error;
+
+}
+
+otError NcpBase::SetPropertyHandler_THREAD_MGMT_PENDING_DATASET(void)
+{
+   otError error = OT_ERROR_NONE;
```
Comment (jwhui) [inline src/ncp/ncp_base_ftd.cpp]: nit: fix indent

**Pull request merged**